lib/model.js

```diff
@@ -3715,7 +3715,8 @@ Model.applyDefaults = function applyDefaults(doc) {
  * @api public
  */
 
-Model.castObject = function castObject(obj) {
+Model.castObject = function castObject(obj, options) {
+  const opts = options || {};
```

Comment:
Creating another variable that points at the same object makes it a bit confusing with no value added. 
```suggestion
  options = options || {};
```